src/datasets/download/streaming_download_manager.py

```diff
@@ -529,126 +505,153 @@ def xglob(urlpath, *, recursive=False, use_auth_token: Optional[Union[str, bool]
         return ["::".join([f"{fs.protocol}://{globbed_path}"] + rest_hops) for globbed_path in globbed_paths]
 
 
-def xpathglob(path, pattern, use_auth_token: Optional[Union[str, bool]] = None):
-    """Glob function for argument of type :obj:`~pathlib.Path` that supports both local paths end remote URLs.
+def xwalk(urlpath, use_auth_token: Optional[Union[str, bool]] = None):
+    """Extend `os.walk` function to support remote files.
 
     Args:
-        path (:obj:`~pathlib.Path`): Calling Path instance.
-        pattern (:obj:`str`): Pattern that resulting paths must match.
+        urlpath (:obj:`str`): URL root path.
+        use_auth_token (:obj:`bool` or :obj:`str`, optional): Whether to use token or token to authenticate on the
+            Hugging Face Hub for private remote files.
 
     Yields:
-        :obj:`~pathlib.Path`
+        :obj:`tuple`: 3-tuple (dirpath, dirnames, filenames).
     """
-    posix_path = _as_posix(path)
-    main_hop, *rest_hops = posix_path.split("::")
+    main_hop, *rest_hops = str(urlpath).split("::")
     if is_local_path(main_hop):
-        yield from Path(main_hop).glob(pattern)
+        yield from os.walk(main_hop)
     else:
-        # globbing inside a zip in a private repo requires authentication
+        # walking inside a zip in a private repo requires authentication
         if rest_hops and (rest_hops[0].startswith("http://") or rest_hops[0].startswith("https://")):
             url = rest_hops[0]
             url, kwargs = _prepare_http_url_kwargs(url, use_auth_token=use_auth_token)
             storage_options = {"https": kwargs}
-            posix_path = "::".join([main_hop, url, *rest_hops[1:]])
+            urlpath = "::".join([main_hop, url, *rest_hops[1:]])
         else:
             storage_options = None
-        fs, *_ = fsspec.get_fs_token_paths(xjoin(posix_path, pattern), storage_options=storage_options)
-        # - If there's no "*" in the pattern, get_fs_token_paths() doesn't do any pattern matching
-        #   so to be able to glob patterns like "[0-9]", we have to call `fs.glob`.
-        # - Also "*" in get_fs_token_paths() only matches files: we have to call `fs.glob` to match directories.
-        # - If there is "**" in the pattern, `fs.glob` must be called anyway.
-        globbed_paths = fs.glob(xjoin(main_hop, pattern))
-        for globbed_path in globbed_paths:
-            yield type(path)("::".join([f"{fs.protocol}://{globbed_path}"] + rest_hops))
+        fs, *_ = fsspec.get_fs_token_paths(urlpath, storage_options=storage_options)
+        for dirpath, dirnames, filenames in fs.walk(main_hop):
+            yield "::".join([f"{fs.protocol}://{dirpath}"] + rest_hops), dirnames, filenames
 
 
-def xpathrglob(path, pattern, **kwargs):
-    """Rglob function for argument of type :obj:`~pathlib.Path` that supports both local paths end remote URLs.
+class xPath(type(Path())):
```

Comment:
many changes in this file are just about moving functions inside this class.

For example I moved xpathrglob to xPath.rglob